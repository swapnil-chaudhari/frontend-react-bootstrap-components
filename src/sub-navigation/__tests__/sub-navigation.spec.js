import React from 'react';
import renderShallow from 'render-shallow';
import SubNavigation from '../sub-navigation';
import SubNavigationItem from '../sub-navigation-item';
import noop from 'src/utils/noop';

describe('<SubNavigation>', () => {
    describe('when it renders', () => {
        let component;

        beforeAll(() => {
            component = renderShallow(
                <SubNavigation onSelect={ noop }>
                    <SubNavigationItem href="project/1" className="active">
                        Project #1
                    </SubNavigationItem>
                    <SubNavigationItem href="project/2">
                        Project #2
                    </SubNavigationItem>
                </SubNavigation>
            ).output;
        });

        test('should have proper markup', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
