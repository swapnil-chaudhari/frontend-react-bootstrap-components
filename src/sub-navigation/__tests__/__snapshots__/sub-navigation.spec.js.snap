// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubNavigation> when it renders should have proper markup 1`] = `
<div
  className="SubNavigation"
>
  <nav
    className="main-subnav"
  >
    <SubNavigationItem
      className="active"
      href="project/1"
      onClick={[Function]}
    >
      Project #1
    </SubNavigationItem>
    <SubNavigationItem
      href="project/2"
      onClick={[Function]}
    >
      Project #2
    </SubNavigationItem>
  </nav>
</div>
`;
